hedgehog/src/Hedgehog/Internal/Runner.hs

```diff
@@ -245,25 +225,27 @@ checkNamed region mcolor name prop = do
 -- | Check a property.
 --
 check :: MonadIO m => Property -> m Bool
-check prop =
+check prop = do
+  mcolor <- detectColor
   liftIO . displayRegion $ \region ->
-    (== OK) . reportStatus <$> checkNamed region Nothing Nothing prop
+    (== OK) . reportStatus <$> checkNamed region mcolor Nothing prop
 
 -- | Check a property using a specific size and seed.
 --
 recheck :: MonadIO m => Size -> Seed -> Property -> m ()
 recheck size seed prop0 = do
+  mcolor <- detectColor
   let prop = withTests 1 prop0
   _ <- liftIO . displayRegion $ \region ->
-    checkRegion region Nothing Nothing size seed prop
+    checkRegion region mcolor Nothing size seed prop
   pure ()
 
 -- | Check a group of properties using the specified runner config.
 --
-checkGroup :: MonadIO m => RunnerConfig -> Group -> m Bool
-checkGroup config (Group group props) =
+checkGroup :: MonadIO m => Bool -> Group -> m Bool
```

Comment:
Could we get something like `data Parallelism = UseParallelism | NoParallelism` here instead of the Bool?